packages/@aws-cdk-testing/framework-integ/test/aws-rds/test/integ.cluster-data-api-to-imported-cluster.ts

```diff
@@ -0,0 +1,35 @@
+import * as cdk from 'aws-cdk-lib/core';
+import * as integ from '@aws-cdk/integ-tests-alpha';
+import * as ec2 from 'aws-cdk-lib/aws-ec2';
+import * as rds from 'aws-cdk-lib/aws-rds';
+import * as lambda from 'aws-cdk-lib/aws-lambda';
+
+const app = new cdk.App();
+
+const stack = new cdk.Stack(app, 'cluster-data-api-to-imported-cluster');
+const vpc = new ec2.Vpc(stack, 'VPC');
+
+const func = new lambda.Function(stack, 'Function', {
+  runtime: lambda.Runtime.NODEJS_18_X,
+  handler: 'index.handler',
+  code: lambda.Code.fromInline('exports.handler = async (event) => { return "hello"; }'),
+});
+
+const cluster = new rds.DatabaseCluster(stack, 'Database', {
+  engine: rds.DatabaseClusterEngine.auroraPostgres({ version: rds.AuroraPostgresEngineVersion.VER_16_1 }),
+  writer: rds.ClusterInstance.serverlessV2('writerInstance'),
+  vpc,
+  enableDataApi: true,
```

Comment:
I wonder what would happen if this is spcified as `false`, but user set it to true when setting `enableDataApi` to `true` on imported interface.

Reply:
Regardless of the actual configuration, the target (in this case Lambda) will be granted the permission if it is set to true on the imported interface. This is true for all attribute values that are set without a lookup (context) method, regardless of the DataAPI. In this case, `secret`, etc.

Another way is to remove that property (`dataApiEnabled`) from the imported interface and make the `enableDataApi` variable undefined in `ImportedDatabaseCluster` class as follows.

```diff
-    this.enableDataApi = attrs.dataApiEnabled ?? false;
```

https://github.com/go-to-k/aws-cdk/blob/a0d9a65d9c699f5c7e8f53ce49de597ce5352d36/packages/aws-cdk-lib/aws-rds/lib/cluster.ts#L999

Then the following branches can be avoided.

```ts
  public grantDataApiAccess(grantee: iam.IGrantable): iam.Grant {
    if (this.enableDataApi === false) {
      throw new Error('Cannot grant Data API access when the Data API is disabled');
    }
```

This would behave in the same way as above. However, it is a little simpler with fewer properties in the imported interface, but it is a little safer with `dataApiEnabled` on the interface.